tools/accuracy_checker/accuracy_checker/adapters/detection.py

```diff
@@ -705,3 +705,148 @@ def _find_output(predictions):
+                        cy = (i + 0.5) * steps[k] / image_size[0]
+                        anchors.append([cx, cy, s_kx, s_ky])
+        anchors = np.clip(anchors, 0, 1)
+
+        return anchors
+
+    def process(self, raw, identifiers=None, frame_meta=None):
+        """
+        Args:
+            identifiers: list of input data identifiers
+            raw: output of model
+        Returns:
+            list of DetectionPrediction objects
+        """
+
+        raw_outputs = self._extract_predictions(raw, frame_meta)
+
+        batch_scores = raw_outputs[self.scores_out]
+        batch_boxes = raw_outputs[self.boxes_out]
+
+        result = []
+        for identifier, scores, boxes, meta in zip(identifiers, batch_scores, batch_boxes, frame_meta):
+            detections = {'labels': [], 'scores': [], 'x_mins': [], 'y_mins': [], 'x_maxs': [], 'y_maxs': []}
+            image_info = meta.get("image_info")[0:2]
+
+            # Prior boxes
+            feature_maps = [[math.ceil(image_info[0] / step), math.ceil(image_info[1] / step)] for step in
+                            self.steps]
+            prior_data = self.prior_boxes(feature_maps, self.min_sizes, image_info, self.steps)
+
+             # Boxes
+            boxes[:, :2] = self.variance[0] * boxes[:, :2]
+            boxes[:, 2:] = self.variance[1] * boxes[:, 2:]
+            boxes[:, :2] = boxes[:, :2] * prior_data[:, 2:] + prior_data[:, :2]
+            boxes[:, 2:] = np.exp(boxes[:, 2:]) * prior_data[:, 2:]
+
+            for label, score in enumerate(np.transpose(scores)):
+
+                # Skip background label
+                if label == 0:
+                    continue
+
+                # Filter out detections with score < confidence_threshold
+                mask = score > self.confidence_threshold
+                filtered_boxes, filtered_score = boxes[mask, :], score[mask]
+                if filtered_score.size == 0:
+                    continue
+
+                # Transform to format (x_min, y_min, x_max, y_max)
+                x_mins = (filtered_boxes[:, 0] - 0.5 * filtered_boxes[:, 2])
+                y_mins = (filtered_boxes[:, 1] - 0.5 * filtered_boxes[:, 3])
+                x_maxs = (filtered_boxes[:, 0] + 0.5 * filtered_boxes[:, 2])
+                y_maxs = (filtered_boxes[:, 1] + 0.5 * filtered_boxes[:, 3])
+
+                # Apply NMS
+                keep = NMS.nms(x_mins, y_mins, x_maxs, y_maxs, filtered_score, self.nms_threshold,
+                               include_boundaries=False, keep_top_k=self.keep_top_k)
+
+                filtered_score = filtered_score[keep]
+                x_mins = x_mins[keep]
+                y_mins = y_mins[keep]
+                x_maxs = x_maxs[keep]
+                y_maxs = y_maxs[keep]
+
+                # Keep topK
+                # Applied just after NMS - no additional sorting is required for filtered_score array
+                filtered_score = filtered_score[:self.keep_top_k]
```

Comment:
what if you have less that keep_top_k boxes after nms?

Reply:
Added check for size of boxes before keep top k